src/index.js

```diff
@@ -609,12 +610,39 @@ const members = {
       const getHtmlNodeOuterHtml = (node) => node.outerHTML;
       const getOuterHtmlByNodeType = {
         1: getHtmlNodeOuterHtml,
-        3: getEscapedTextContent,
         8: getCommentNodeOuterHtml,
       };
+      // Text nodes with these ancestors should be treated as raw text
+      // See section 8.4 of
+      // https://www.w3.org/TR/2008/WD-html5-20080610/serializing.html#html-fragment
+      const rawTextNodeAncestors = new Set([
```

Comment:
Will change this to an object instead, as Set is not support in Opera, and I don't want to introduce another dependency just for that.